File: sandbox/src/resources/stats.ts
import type { Stats, StatsQuery } from '../types';
import type { RequestOptions } from '../transport/http';
import { HttpTransport } from '../transport/http';

export class StatsResource {
  private readonly transport: HttpTransport;
  private readonly sandboxId: string;

  /** @internal Create the stats wrapper for one sandbox. */
  public constructor(transport: HttpTransport, sandboxId: string) {
    this.transport = transport;
    this.sandboxId = sandboxId;
  }

  /** Fetch sandbox usage stats for a lookback window. */
  public stats(query: StatsQuery = {}, options?: RequestOptions): Promise<Stats> {
    const params = new URLSearchParams();

    if (query.hoursAgo !== undefined) {
      params.set('hoursAgo', String(query.hoursAgo));
    }

    return this.transport.requestJson<Stats>({
      endpoint: `/sandboxes/${this.sandboxId}/stats`,
      method: 'GET',
      query: params,
      ...options,
    }) as Promise<Stats>;
  }
}

File: sandbox-python/brimble_sandbox/resources/stats.py
"""Sandbox metrics operations."""

from __future__ import annotations

from ..transport import HttpTransport, RequestOptions, RetryOptions
from ..types import Stats, StatsQuery


class StatsResource:
    """Fetch stats for one sandbox."""

    def __init__(self, transport: HttpTransport, sandbox_id: str) -> None:
        self._transport = transport
        self._sandbox_id = sandbox_id

    def stats(
        self,
        query: StatsQuery | None = None,
        *,
        timeout_ms: int | None = None,
        retry: RetryOptions | bool | None = None,
    ) -> Stats:
        """Fetch CPU/memory/network usage stats."""
        options = RequestOptions(timeout_ms=timeout_ms, retry=retry)
        return self._transport.request_json(
            endpoint=f"/sandboxes/{self._sandbox_id}/stats",
            method="GET",
            query=query or {},
            options=options,
        )
